app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,70 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.exceptions.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.repository.TaskRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin("http://localhost:3000")
+public class TaskController {
+    TaskRepository taskRepository = new TaskRepository();
+
+    @GetMapping
+    public List<Task> list() {
+        return taskRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Task detail(@PathVariable Long id) {
+        if (taskRepository.isExist(id)) {
+            return taskRepository.findById(id);
+        }
+        throw new TaskNotFoundException();
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Task create(@RequestBody Task task) {
+        return taskRepository.save(task);
+    }
+
+    @RequestMapping(path = "/{id}", method = {RequestMethod.PUT, RequestMethod.PATCH})
+    public Task update(@PathVariable Long id, @RequestBody Task task) {
+        if (taskRepository.isExist(id)) {
+            task.setId(id);
+            return taskRepository.update(task);
+        }
+        throw new TaskNotFoundException();
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void delete(@PathVariable Long id) {
+        if (taskRepository.isExist(id)) {
+            taskRepository.delete(id);
+            return;
+        }
+        throw new TaskNotFoundException();
```

Comment:
`return;` 이라는 코드가 이질적으로 느껴지네요! 

조금만 코드가 더 많아지면 코드들이 어디서 멈추고 어디서는 계속 진행되는지 헷갈릴 수 있어요! 

```suggestion
        if (!taskRepository.isExist(id)) {
            throw new TaskNotFoundException();
         }
```

이런식으로 빠른 실패를 하는 코드를 위로 옮겨보는 것은 어떨까요?

Reply:
7f4875bf4c9806f9d9864446673a297f721d1f77 옮겼습니다!